src/algopy_testing/context.py

```diff
@@ -685,11 +706,29 @@ def any_application(  # type: ignore[misc]
         new_app_id = id if id is not None else next(self._app_id)
         new_app = algopy.Application(new_app_id)
 
-        if address is None:
-            address = algopy.Account(algosdk.logic.get_application_address(new_app_id))
+        # Set sensible defaults
+        default_app_fields = {
+            "approval_program": self.any_bytes(64),
```

Comment:
rather than any bytes, might be better to just use a constant bytes representing an [always approve program](https://github.com/algorandfoundation/puya/blob/main/test_cases/inner_transactions/programs.py#L7), since random bytes are unlikely to be a valid program